src/modules/hyprland/workspaces.cpp

```diff
@@ -448,7 +447,7 @@ void Workspaces::sort_workspaces() {
                   return is_name_less;
                 case SORT_METHOD::NUMBER:
                   try {
-                    return is_number_less;
+                    return std::stoi(a->name()) < std::stoi(b->name());
```

Comment:
I think this can be improved more by changing the 3 helper comparisons to be lambdas instead of boolean variables. 

Like this:

```cpp
auto is_id_less = [a, b]() { a->id() < b->id(); };
...
```

We can then easilly use: 

`return is_number_less()` (with the brackets, because it's a function)

Reply:
sure. I'm a bit rusty on c++ so i'm quite slow, but i want to make a PR to the Hyprland project enabling persistent workspaces from the compositor itself (if somebody don't beat me to it, it's all very active!), so i'll probably also take a better look at this part of waybar if there is space for improvement. 
That fix I made was done in a real rush!